packages/firebase_core/firebase_core_web/lib/src/firebase_core_web.dart

```diff
@@ -14,23 +14,44 @@ class FirebaseWebService {
   /// property allows overriding of web naming to Flutterfire plugin naming.
   String? override;
 
+  /// Function to call to ensure the Firebase Service is initalized.
+  /// Usually used to ensure that the Web SDK match the behavior
+  /// of native SDKs.
+  EnsureInitializedFunction ensureInitializedFunction;
+
   /// Creates a new [FirebaseWebService].
-  FirebaseWebService._(this.name, [this.override]);
+  FirebaseWebService._(
+    this.name, {
+    this.override,
+    this.ensureInitializedFunction,
```

Comment:
Maybe change`ensureInitializedFunction` to `ensurePluginInitialized`.

Reply:
Changed 👍 Will probably be clearer :)